src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs

```diff
@@ -689,49 +863,202 @@ private static IEnumerable<double> LazySequence(double records, double start, do
             }
         }
 
-        public static FormulaValue Abs(IRContext irContext, NumberValue[] args)
+        private static IEnumerable<decimal> LazySequenceDecimal(int records, decimal start, decimal step)
+        {
+            var x = start;
+            for (var i = 1; i <= records; i++)
+            {
+                yield return x;
+                x += step;
+            }
+        }
+
+        public static FormulaValue Abs(IRContext irContext, FormulaValue[] args)
         {
             var arg0 = args[0];
-            var x = arg0.Value;
-            var val = Math.Abs(x);
+
+            if (arg0 is NumberValue num)
+            {
+                return AbsFloat(irContext, num);
+            }
+            else if (arg0 is DecimalValue dec)
+            {
+                return AbsDecimal(irContext, dec);
+            }
+            else
+            {
+                return CommonErrors.UnreachableCodeError(irContext);
+            }
+        }
+
+        public static FormulaValue AbsFloat(IRContext irContext, NumberValue arg)
+        {
+            double x = arg.Value;
+            double val = Math.Abs(x);
             return new NumberValue(irContext, val);
         }
 
-        public static FormulaValue Round(IRContext irContext, NumberValue[] args)
+        public static FormulaValue AbsDecimal(IRContext irContext, DecimalValue arg)
+        {
+            decimal x = arg.Value;
+            decimal val = x < 0m ? -x : x;

```

Comment:
>x < 0m ? -x : x; [](http://example.com/codeflow?start=26&length=16)

why not using decimal.Abs(x)?

Reply:
Interestingly, decimal does not have an Abs method.

Reply:
My bad, it is actually on the Math object.  Updated.